ctest-next/templates/test.rs

```diff
@@ -82,7 +83,430 @@ mod generated_tests {
+                    __test_offset_{{ ident }}_{{ field.ident() }}() as usize,
+                    "field offset {{ field.ident() }} of {{ ident }}");
+            check_same(mem::size_of_val(&val) as u64,
+                    __test_fsize_{{ ident }}_{{ field.ident() }}(),
+                    "field size {{ field.ident() }} of {{ ident }}");
+        }
+        {%- if !self::should_skip_field_type(generator, Either::Left(structure), field) +%}
+
+        extern "C" {
+            fn __test_field_type_{{ ident }}_{{ field.ident() }}(a: *mut {{ ident }}) -> *mut u8;
+        }
+        unsafe {
+            let mut uninit_ty = std::mem::MaybeUninit::<{{ ident }}>::uninit();
+            let uninit_ty = uninit_ty.as_mut_ptr();
+            let ty_ptr_mut = std::ptr::addr_of_mut!(*uninit_ty);
+            let field_ptr = std::ptr::addr_of!((*uninit_ty).{{ field.ident() }});
+            check_same(field_ptr as *mut _,
+                    __test_field_type_{{ ident }}_{{ field.ident() }}(ty_ptr_mut),
+                    "field type {{ field.ident() }} of {{ ident }}");
+            #[allow(unknown_lints, forgetting_copy_types)]
+            mem::forget(uninit_ty);
+        }
+        {%- endif +%}
+        {%- endif +%}
+        {%- endfor +%}
+    }
+
+    {%- if self::should_roundtrip(generator, ident) +%}
+
+    /// Generates a padding map for a specific type.
+    ///
+    /// Essentially, it returns a list of bytes, whose length is equal to the size of the type in
+    /// bytes. Each element corresponds to a byte and has two values. `1` if the byte is padding,
+    /// and `0` if the byte is not padding.
+    ///
+    /// For type aliases, the padding map is all zeroes.
+    fn roundtrip_padding_{{ ident }}() -> Vec<u8> {
+        // stores (offset, size) for each field
+        {#- If there is no public field these become unused. +#}
+        {%- if fields.len() > 0 +%}
+        let mut v = Vec::<(usize, usize)>::new();
+        let bar = std::mem::MaybeUninit::<{{ ident }}>::uninit();
+        let bar = bar.as_ptr();
+        {%- else +%}
+        let v = Vec::<(usize, usize)>::new();
+        {%- endif +%}
+        {%- for field in fields +%}
+        unsafe {
+            let ty_ptr = std::ptr::addr_of!((*bar).{{ field.ident() }});
+            let val = ty_ptr.read_unaligned();
+            let size = mem::size_of_val(&val);
+            let off = offset_of!({{ ident }}, {{ field.ident() }});
+            v.push((off, size));
+        }
+        {%- endfor +%}
+        // This vector contains `1` if the byte is padding
+        // and `0` if the byte is not padding.
+        let mut pad = Vec::<u8>::new();
+        // Initialize all bytes as:
+        //  - padding if we have fields, this means that only
+        //  the fields will be checked
+        //  - no-padding if we have a type alias: if this
+        //  causes problems the type alias should be skipped
+        pad.resize(mem::size_of::<{{ ident }}>(), 1);
```

Comment:
Could the type be `Vec<bool>`?

Also, does this even need to be a vector? I think `[true; size_of::<{{ ident }}>]` should work.